src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -71,6 +71,31 @@ describeEachAppLayout(({ theme, size }) => {
     expect(wrapper.findDrawersTriggers()).toHaveLength(2);
   });
 
+  test('update rendered drawers via runtime config', async () => {
+    awsuiPlugins.appLayout.registerDrawer(drawerDefaults);
+    const { wrapper } = await renderComponent(<AppLayout />);
+    // the 2nd trigger is for tools
+    expect(wrapper.findDrawersTriggers()).toHaveLength(2);
+
+    awsuiPlugins.appLayout.updateDrawer({
+      id: 'test',
+      badge: true,
+      ariaLabels: {
+        triggerButton: 'test-trigger-button',
+      },
+    });
+    await delay();
+
+    if (theme !== 'classic' && size === 'desktop') {
+      expect(wrapper.find('[class*=awsui_dot_]')?.getElement()).toBeInTheDocument();
```

Comment:
1. you can import the related styles file and call `triggerStyles.dot` class name directly
2. There need to be two assertions 1) no badge on the first render 2) there is a badge after calling `updateDrawer` and `delay`
3. We need to check if the dot is on the correct toggle, via `wrapper.findDrawerTriggerById(drawerDefaults.id).findByClassName(toggleStyles.dot).getElement()`

Reply:
updated.